src/components/utility/CopyCodeToClipboard/CopyCodeToClipboard.astro

```diff
@@ -0,0 +1,55 @@
+<style lang='scss' is:global>
+	.code-wrapper {
+		display: flex;
+		justify-content: flex-end;
+		right: 16px;
+	}
+
+	.copy-button {
+		transition: background-color 0.2s, color 0.2s;
+		cursor: pointer;
+		padding: 0.8rem 1.8rem;
+		font-weight: bold;
+	}
+
+	.success {
+		background-color: var(--theme-text-accent);
+		color: var(--theme-bg);
+		animation: button-success 0.2s ease-in-out;
+	}
+
+	@keyframes button-success {
+		0% {
+			transform: scale(1);
+		}
+		20% {
+			transform: scale(1.1);
+		}
+		100% {
+			transform: scale(1);
+		}
+	}
+</style>
+
+<script>
+	import {
+		createCopyButton,
+		createCopyButtonWrapper,
+		handleCopyButtonClick,
+	} from './helpers'
+
+	const codeSnippets = [
+		...document.querySelectorAll('.astro-code'),
+	] as HTMLElement[]
```

Comment:
don't need to use type casting

Reply:
Without type casting TS communicates type error

image.png

Reply:
Changed to use generic